diff --git a/src/libvirt-broker.vala b/src/libvirt-broker.vala
index 655a8c562e2acc044ec66b3c04bcb4a05b9e3193..140e09c8669ddfd74ba3c23b91cd92282f6b74b7 100644
--- a/src/libvirt-broker.vala
+++ b/src/libvirt-broker.vala
@@ -24,9 +24,10 @@ public GVir.Connection get_connection (string name) {
         return broker.connections.get (name);
     }
 
-    public async LibvirtMachine add_domain (CollectionSource source, GVir.Connection connection, GVir.Domain domain)
-                                            throws GLib.Error {
-        return_val_if_fail (broker != null, null);
+    public async LibvirtMachine? add_domain (CollectionSource source, GVir.Connection connection, GVir.Domain domain)
+                                             throws GLib.Error {
+        if (broker == null)
+            return null;
 
         if (pending_domains.find (domain) != null) {
             // Already being added asychronously